GIT_CONFLICTS.md

```diff
@@ -0,0 +1,98 @@
+## Common QnA
+
+<details>
+<summary>
+<em>How can I know if I’m in a bad situation?</em>
+</summary>
+
+- Run `gitk` or `gitk --all` to visualize the commit tree.
+- Check if `git log` shows any merge commits, indicating a lack of proper synchronization with the upstream branch.
+</details>
+
+<details>
+<summary>
+<em>How to Hard Sync Master with Upstream</em>
+</summary>
+
+To hard sync your local master branch with the upstream master branch, follow these steps in your main branch:
+
+- `git remote add upstream /url-to-original-repo`
+- `git fetch upstream`
+- `git reset --hard upstream/main`
+- `git push origin main --force`
+</details>
+
+## Better commits
+
+- Avoid committing unnecessary files:
+    - Avoid using `git commit -a`, which commits everything.
+- Use targeted commit commands:
+    - Use `git add` to stage specific files.
+    - Use `git add -p` for a patch mode to review changes before staging.
+- Ensure changes are correct before committing:
+    - Use `git diff --cached` to display staged changes.
+    - Use `git commit -v` to view diffs while writing the commit message.
+
+If you want to make sure about what you’re committing, use `git diff --cached` to display the ready-to-commit changes. Also, I like to use `git commit -v`, which displays the diff as a comment, so that I can view it while writing the commit message.
+
+<details>
+<summary>
+<em>Edit a commit</em>
+</summary>
+
+- `git reset --soft HEAD^`: Reset to the previous commit without changing files.
+- Edit the necessary changes.
+- `git commit -a -v -c ORIG_HEAD`: Recommit with the same message and verify changes.
+- After pushing, use force push: `git push -f`.
+</details>
+
+<details>
+<summary>
+<em>What should I do if I’m in a bad situation?</em>
+</summary>

```

Comment:
Same comment, kindly consider the review comment as whole, and don't expect me to hand hold and to pin point everything, I am not rigid on the content, but i accept the review comments are gracefully accepted and worked on